keanu-python/tests/test_sequence.py

```diff
@@ -191,3 +193,26 @@ def factory2(sequence_item):
     __check_output_equals(sequence, x2_label, 2)
     __check_output_equals(sequence, x3_label, 8)
     __check_output_equals(sequence, x4_label, 0.125)
+
+
+def test_last_item_retrieved_correctly() -> None:
+    x_label = "x"
+    x_input_label = Sequence.proxy_for(x_label)
+
+    def factory(sequence_item):
+        x = DoubleProxy((), x_input_label)
+        x_out = x * Const(2.0)
+        x_out.set_label(x_label)
+        sequence_item.add(x_out)
+        sequence_item.add(x)
```

Comment:
@icantsayrural here is an example of two items, both with the inner label namespace "x", being added to the same sequence item. This is necessary as otherwise the sequence building can't function properly.

Reply:
If `get_contents()` were to return a map with just the inner namespace of vertices as the keys then it's not clear or defined whether the proxy or output vertex would be the one keyed by "x".

Reply:
Nice. Can we check if the contents have `proxy_for.x_label`?

Reply:
Otherwise lgtm!